components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt

```diff
@@ -65,6 +65,18 @@ abstract class EngineSession(
             fun select(vararg categories: Int): TrackingProtectionPolicy =
                 TrackingProtectionPolicy(categories.sum())
         }
+
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (javaClass != other?.javaClass) return false
```

Comment:
I think if you use the following then Kotlin will smart cast and you do not need to do that manually:

```Kotlin
if (other !is TrackingProtectionPolicy) return false
```

Reply:
👍 this was auto-generated by AS. Good find.